### PR TITLE
Fix nodet format option

### PR DESCRIPTION
**Pull request by VadimKovalenkoSNF**
Fix https://github.com/openzim/mwoffliner/issues/1868

### CODE REVIEW

**Review by kelson42 — CHANGES_REQUESTED** [review]
I failed to test it:
```
> mwoffliner@1.13.0 start
> ./node_modules/.bin/ts-node-esm ./src/cli.ts --adminEmail=kelson@kiwix.org --osTmpDir=/dev/shm --optimisationCacheUrl=http://s3.us-west-1.wasabisys.com/?bucketName=org-kiwix-dev-mwoffliner&keyId=.........&secretAccessKey=....... --mwUrl=https://fr.wikipedia.org --format=nodet --articleList=Paris,Berlin --verbose --forceRender=VisualEditor

[log] [2023-09-11T18:06:54.692Z] Starting mwoffliner v1.13.0...
[info] [2023-09-11T18:06:54.715Z] Using custom flavour: no
[log] [2023-09-11T18:06:54.721Z] closing sanitize redis DB
[log] [2023-09-11T18:06:55.036Z] Successfully logged in S3
[log] [2023-09-11T18:06:55.042Z] Getting text direction...
[log] [2023-09-11T18:06:55.043Z] Getting site info...
[log] [2023-09-11T18:06:55.044Z] Getting sub-title...
[info] [2023-09-11T18:06:55.045Z] Downloading [https://fr.wikipedia.org/wiki/]
[info] [2023-09-11T18:06:55.047Z] Getting JSON from [https://fr.wikipedia.org/w/api.php?action=query&meta=siteinfo&format=json&formatversion=2&siprop=general%7Cnamespaces%7Cstatistics%7Cvariables%7Ccategory%7Cwikidesc]
[info] [2023-09-11T18:06:55.049Z] Downloading [https://fr.wikipedia.org/wiki/]
[log] [2023-09-11T18:06:55.305Z] Text direction is [ltr]
[info] [2023-09-11T18:06:55.305Z] Getting JSON from [https://fr.wikipedia.org/w/api.php?action=query&meta=siteinfo&format=json&formatversion=2]
[info] [2023-09-11T18:06:55.457Z] Downloading [https://fr.wikipedia.org/static/images/project-logos/frwiki.png]
[error] [2023-09-11T18:06:55.966Z] Failed to run mwoffliner after [2s]: {
	"stack": "Error: Input file is missing: ",
	"message": "Input file is missing: "
}
[error] [2023-09-11T18:06:55.966Z] 

**********

Input file is missing: 

**********

```

**Comment by VadimKovalenkoSNF** [comment]
@kelson42 
> -forceRender=VisualEditor
fr wiki doesn't support VisualEditor, I got `Cannot create an instance of VisualEditor renderer.` in my logs.

**Comment by VadimKovalenkoSNF** [comment]
> I really wonder about that because VE is activated on all WPs AFAIK?! 

Yes, you're right. After additional debug I found that error occurred because of encoded characters in URL in the French wiki when checking capabilities.
> Why i don't have a proper error?

I reproduced your entire command and also got error `Input file is missing:`. This happened not because VisualRenderer (I removed it from CLI params) but because of problems with `--optimisationCacheUrl`. This error occurs for both En and Fr wiki, it doesn't suppose rely on renderer
> Why this is not check at start with proper exit?

Because capabilities check happens after S3 and Downloader instantiation. This is definitely something with S3. My previous comment was about French wiki and VisualEditor, because I didn't expect to get any other error, so I didn't include  `--optimisationCacheUrl` param to the URL.

**Comment by VadimKovalenkoSNF** [comment]
Upd: error with `Input file is missing:` happened right after [migration](https://github.com/openzim/mwoffliner/pull/1865) to awd-sdk v3.

File: src/util/mw-api.ts [hunk 1]
```diff
@@ -259,7 +259,7 @@ export function mwRetToArticleDetail(obj: QueryMwRet): KVS<ArticleDetail> {
 
 export async function checkApiAvailability(url: string, loginCookie = ''): Promise<boolean> {
   try {
-    const resp = await axios.get(url, { maxRedirects: 0, headers: { cookie: loginCookie } })
+    const resp = await axios.get(decodeURI(url), { maxRedirects: 0, headers: { cookie: loginCookie } })
```
Comment (VadimKovalenkoSNF) [inline src/util/mw-api.ts:262]: Decoding URL should help to avoid problems with querying to MW Action API for different foreign wikis.

**Comment by kelson42** [comment]
Hard to test without having #1904 first fixed.

**Review by kelson42 — CHANGES_REQUESTED** [review]
I have tested with `--mwUrl="https://fr.wikipedia.org" --format=nodet --verbose --articleList="Berlin,Paris" --forceRender=VisualEditor`. Paris article is OK but suprisingly Berlin is not.

Work as expected with `WikimediaDesktop` but not really with `VisualEditor` render. I expect such PR to integrate automated tests on all supported end-points systematicaly.

**Comment by VadimKovalenkoSNF** [comment]
> Work as expected with WikimediaDesktop but not really with VisualEditor render

I could not reproduce it on the rebased version of this patch, works as expected on my side (both articles scraped correctly). If you face this issue again, please give me logs.

> I expect such PR to integrate automated tests on all supported end-points systematically.

There is a dedicated [test](https://github.com/openzim/mwoffliner/pull/1902/files#diff-8dc3073e26ab6fbbd6354798464a32fa777588212b918693f64c9693fb7b43dfR82) for VE and nodet

**Review by kelson42 — APPROVED** [review]
No it works?! So LGTM, but the test will have to be rewritten in an appropriate manner, so much code duplication is NOK

**Pull request merged**